src/world/Manager/MapMgr.cpp

```diff
@@ -0,0 +1,889 @@
+#include <Common.h>
+#include <Service.h>
+
+#include <datReader/DatCategories/bg/lgb.h>
+#include <Exd/ExdDataGenerated.h>
+
+#include <Event/EventHandler.h>
+
+#include <Network/CommonActorControl.h>
+#include <Network/GameConnection.h>
+#include <Network/PacketDef/Zone/ServerZoneDef.h>
+#include <Network/PacketWrappers/ActorControlSelfPacket.h>
+
+#include <Logging/Logger.h>
+#include <Util/Util.cpp>
+
+#include <Actor/Player.h>
+#include <Inventory/Item.h>
+
+#include "MapMgr.h"
+#include "TerritoryMgr.h"
+
+#include "ServerMgr.h"
+
+using namespace Sapphire::Event;
+using namespace Sapphire::Network::Packets;
+using namespace Sapphire::Network::Packets::Server;
+
+Sapphire::World::Manager::MapMgr::MapMgr()
+{  
+  auto& exdData = Common::Service< Data::ExdDataGenerated >::ref();
+
+  size_t count = 0;
+
+  for( auto quest : exdData.m_QuestDat.get_rows() )
+  {
+    if( exdData.getField< std::string >( quest.second, 1 ).empty() ) // id
+      continue;
+
+    auto& questData = m_questData[ quest.first ];
```

Comment:
I've tried it in this way first, but it takes approximately 10-20 ms to load the quest data directly from the exd. This is way too long, if you want to do it after the ZoneIn trigger like the retail server does.
Even if I remove the unused vectors from the Quest struct, it isn't really faster.
There are zones which contains over 100 quests to check. That's over a second processing time.
I didn't found any other way to this faster without storing them in memory.

Reply:
ah yeah, okay, i'm not surprised, datreader needs more than a bit of love in some areas

you can keep them in memory though, its just loading everything upfront is a bit costly when working on the server - i don't mind it that much as when i tried it, it wasn't too bad but its nice to avoid doing everything upfront where reasonable